tests/test_requests.py

```diff
@@ -526,6 +526,19 @@ def test_proxy_error(self):
         with pytest.raises(ProxyError):
             requests.get('http://localhost:1', proxies={'http': 'non-resolvable-address'})
 
+    def test_proxy_error_on_bad_url(self, httpbin, httpbin_secure):
+        with pytest.raises(ProxyError):
+            requests.get(httpbin_secure(), proxies={'https': 'http:/badproxyurl:3128'})
+
+        with pytest.raises(ProxyError):
+            requests.get(httpbin(), proxies={'http': 'http://:8080'})
+
+        with pytest.raises(ProxyError):
+            requests.get(httpbin_secure(), proxies={'https': 'https://'})
+
+        with pytest.raises(ProxyError):
+            requests.get(httpbin(), proxies={'http': 'http:///idontexist:8080'})
```

Comment:
Sorry, one more note. It would probably be good to make sure we're testing with a more common looking host too. Just so we're disambiguating between what could be a valid path (`/idontexist`).
e.g.
```Python
http:///example.com:8080
```

Reply:
Updated.